src/com/lilithsthrone/game/sex/sexActions/baseActionsMisc/GenericOrgasms.java

```diff
@@ -1220,40 +1220,103 @@ private static String cumTargetDescription(GameCharacter characterOrgasming, Gam
-						switch(characterOrgasming.getCumFlavour()) {
-							case BEER:
-								cumTargetSB.append(" The unusual taste of [npc1.name]'s beer-flavoured cum rises up to hit your [npc2.tongue], and you're left with no other option but to gulp down as much of the");
-								break;
-							case CHOCOLATE:
-								cumTargetSB.append(" The unusual taste of [npc1.name]'s chocolate-flavoured cum rises up to hit your [npc2.tongue], and you're left with no other option but to gulp down as much of the");
-								break;
-							case CUM:
-								cumTargetSB.append(" The salty taste of cum rises up to hit your [npc2.tongue], and you're left with no other option but to gulp down as much of the");
-								break;
-							case GIRL_CUM:
-								cumTargetSB.append(" The unusual taste of [npc1.name]'s sweet cum rises up to hit your [npc2.tongue], and you're left with no other option but to gulp down as much of the");
-								break;
-							case HONEY:
-								cumTargetSB.append(" The unusual taste of [npc1.name]'s honey-flavoured cum rises up to hit your [npc2.tongue], and you're left with no other option but to gulp down as much of the");
-								break;
-							case MILK:
-								cumTargetSB.append(" The unusual taste of [npc1.name]'s milk-flavoured cum rises up to hit your [npc2.tongue], and you're left with no other option but to gulp down as much of the");
-								break;
-							case MINT:
-								cumTargetSB.append(" The unusual taste of [npc1.name]'s mint-flavoured cum rises up to hit your [npc2.tongue], and you're left with no other option but to gulp down as much of the");
-								break;
-							case PINEAPPLE:
-								cumTargetSB.append(" The unusual taste of [npc1.name]'s pineapple-flavoured cum rises up to hit your [npc2.tongue], and you're left with no other option but to gulp down as much of the");
-								break;
-							case SLIME:
-								cumTargetSB.append(" The unusual taste of [npc1.name]'s sweet cum rises up to hit your [npc2.tongue], and you're left with no other option but to gulp down as much of the");
-								break;
-							case STRAWBERRY:
-								cumTargetSB.append(" The unusual taste of [npc1.name]'s strawberry-flavoured cum rises up to hit your [npc2.tongue], and you're left with no other option but to gulp down as much of the");
-								break;
-							case VANILLA:
-								cumTargetSB.append(" The unusual taste of [npc1.name]'s vanilla-flavoured cum rises up to hit your [npc2.tongue], and you're left with no other option but to gulp down as much of the");
-								break;
+						if(target.hasFetish(Fetish.FETISH_CUM_ADDICT) || SexFlags.playerRequestedCreampie)
+						{
+							switch(characterOrgasming.getCumFlavour()) {
+								case BEER:
+									cumTargetSB.append(" The unusual taste of [npc1.name]'s beer-flavoured cum rises up to hit your [npc2.tongue], and you "
```

Comment:
These switch statements seem to end in the same format so how about something like this

```java
switch(characterOrgasming.getCumFlavour()) {
    case BEER:
        cumTargetSB.append(" The unusual taste of [npc1.name]'s beer-flavoured cum rises up to hit your [npc2.tongue], and you");
    // all other cases
}

if(target.hasFetish(Fetish.FETISH_CUM_ADDICT) || SexFlags.playerRequestedCreampie) {
    cumTargetSB.append(" " + UtilText.returnStringAtRandom("greedily","hungrily") + " gulp down as much of the");
else {
    cumTargetSB.append("'re left with no other option but to gulp down as much of the");
}

Reply:
That would fragment the lines more, though. I worry that that would make it hard to follow.

Reply:
I agree with pinnae here; the lines are already quite fragmented.

If you want to de-frag the flow, I'd suggest splitting it out into a separate method for each chunk, then make cumTargetDescription() just stitch things together. You might have to do this for other descriptions, to make things consistent, though.